infrastructures/infrastructure-factory/src/main/java/org/eclipse/che/api/factory/server/scm/kubernetes/KubernetesPersonalAccessTokenManager.java

```diff
@@ -186,6 +188,7 @@ private Optional<PersonalAccessToken> doGetPersonalAccessToken(
               && (oAuthProviderName == null
                   || oAuthProviderName.equals(
                       annotations.get(ANNOTATION_SCM_PERSONAL_ACCESS_TOKEN_NAME)))
+              && (!isNullOrEmpty(trimmedUrl)) // ignore PAT with empty server endpoint URL
```

Comment:
The `ANNOTATION_SCM_URL` is mandatory so I think it is better to move this check to the beginning of the loop body: 
```java
if (isNullOrEmpty(trimmedUrl)) {
   continue;
}
```